Runtime/JSArray.Proxy.cs

```diff
@@ -0,0 +1,176 @@
+using System.Collections.Generic;
+
+namespace NodeApi;
+
+public partial struct JSArray
```

Comment:
>JSArray [](http://example.com/codeflow?start=22&length=7)

I just wonder why these proxies are part of JSArray? I would expect them to be completely independent classes/structs. There seemed to be no relationship with JSArray.

Reply:
The `Proxy()` methods that take an `IList<T>` or `IReadOnlyList<T>` actually do use a `JSArray` as the proxy target. The `Proxy()` methods that take an `ICollection<T>` or `IReadOnlyCollection<T>` use a plain object instead of an array. I didn't have a better place to put those methods because there is no JS "collection" class.

Note there is also a `Proxy()` method on `JSIterable`, and later I'll make similar ones for `JSSet` and `JSMap` that proxy the corresponding interfaces. It's only the "collection" proxies that don't have a good home.